test/acceptance/tests/mesh-gateway/mesh_gateway_test.go

```diff
@@ -107,6 +107,16 @@ func TestMeshGatewayDefault(t *testing.T) {
 	logger.Log(t, "verifying federation was successful")
 	verifyFederation(t, primaryClient, secondaryClient, releaseName, false)
 
+	// Log services in DC2 that DC1 is aware of before exiting this test
+	defer func() {
+		svcs, _, err := primaryClient.Catalog().Services(&api.QueryOptions{Datacenter: "dc2"})
+		if err != nil {
+			logger.Logf(t, "error calling primary on /v1/catalog/services?dc=dc2: %s\n", err.Error())
+			return
+		}
+		logger.Logf(t, "primary on /v1/catalog/services?dc=dc2: %+v\n", svcs)
```

Comment:
wasn't sure if `logger` is still valid if the test fails

Reply:
I think it should log if the test fails, at least in failing tests i've seen logs like "creating static-server in dc2" which is from L121 below using `logger`